Source/Other/Extensions/BundleExtension.swift

```diff
@@ -20,6 +22,77 @@ import Foundation
 		return nil
 	}
 
+    public var version: String? {
+
+        if let info = self.infoDictionary {
+            if let version = info["CFBundleShortVersionString"]{
+                return version as? String
+            }
+        }
+
+        return nil
+    }
+
+    public var bundleIdentifier: String? {
+
+        if let info = self.infoDictionary {
+            if let bundleIdentifier = info[kCFBundleIdentifierKey as String]{
+                return bundleIdentifier as? String
+            }
+        }
+
+        return nil
+    }
+
+    public var build: String? {
+
+        if let info = self.infoDictionary {
+            if let build = info[kCFBundleVersionKey as String]{
+                return build as? String
+            }
+        }
+
+        return nil
+    }
+
+    public var isSnapshotBuild: Bool {
+
+        guard let ret = appName?.contains(SPSnapshotBuildIndicator)
+        else{
+            return false
+        }
+
+        return ret
+    }
+
+    public func updateAvailable() -> Dictionary<String, String> {
+        let url = URL(string: "https://itunes.apple.com/lookup?bundleId=\(bundleIdentifier ?? "")")
```

Comment:
Two other stretch goal things - we can tell the currentVersionReleaseDate from that response. Perhaps we should wait x days after a new update is approved before prompting users to install it?

Also, we can pull the release notes out of the response too (`releaseNotes`) and those may also be very nice to show in the popup

Reply:
Last thing (sorry for all the comments) - `minimumOsVersion` also comes out of this response. I think perhaps that is one we should check too and not prompt the user if their macOS version is older than our latest release's min macOS version

Reply:
Doesn't look great on Big Sur (first attempt)

<img width="260" alt="Screenshot 2021-02-11 at 4 21 34 PM" src="https://user-images.githubusercontent.com/1576190/107614732-d8485000-6c85-11eb-8985-dbf70fcfd7a3.png">

Reply:
I might be the only one against this, but I think we should go different path...

This modal is mostly important for non-AppsStore users - so I would change "Go to Mac App Store" to "Open Github" and open github releases page. Opening Mac AppStore is one click for users, going to github is more complicated. AppStore users then don't need a button.

I don't think we need the release notes. In case of bigger releases, it will look terrible. Can't think of any program that would show it right away.

I think we should show this alert, as a title have "New update of Sequel Ace is available!" and then have buttons "Open Github", "Not now" and then checkmark "Ignore this version".

On top of that, in preferences we should have checkmark "Show alert when update is available" that user could disable when he doesn't want to be notified at all.

Reply:
a little better with an accessory view:

<img width="266" alt="Screenshot 2021-02-11 at 9 11 59 PM" src="https://user-images.githubusercontent.com/1576190/107640933-f033ca00-6cad-11eb-875e-28a0de932293.png">

Reply:
Not so good on 10.15:

<img width="423" alt="Screenshot 2021-02-11 at 9 23 19 PM" src="https://user-images.githubusercontent.com/1576190/107642416-d1363780-6caf-11eb-8bd2-eca8c326c67d.png">

Reply:
Yeah I still think what I said. :)

Reply:
> This modal is mostly important for non-AppsStore users - so I would change "Go to Mac App Store" to "Open Github" and open github releases page. Opening Mac AppStore is one click for users, going to github is more complicated. AppStore users then don't need a button.

Makes more sense. Thoughts @Jason-Morcos?

Reply:
I think that makes sense, could we check the executable to see which way they downloaded it and switch on that? We should be able to poll the receipt for the purchase and I think we could use that to determine if they're an App Store or other (GitHub) download?

Reply:
Possibly helpful - https://www.macobserver.com/tmo/article/understanding_mac_app_store_app_restrictions

Reply:
I'm still not sure if we want to do it. We have phased rollout on purpose, why would we show this to everyone on 1%? I would just... ignore AppStore.

Reply:
That's fair, if so, we should still probably poll for the App Store receipt to disable this prompt if they downloaded from the App Store, right? My point is showing a prompt to go to GitHub for users who downloaded from the App Store is confusing and backwards.

Reply:
`Release notes` instead of `Open Github` would solve it 😛 Anyways I'm definitely not a blocker here, just coming with ... ideas :)

Reply:
A GitHub release checker version. ***WIP***

https://github.com/Sequel-Ace/Sequel-Ace/tree/github-release-checker

Reply:
> That's fair, if so, we should still probably poll for the App Store receipt to disable this prompt if they downloaded from the App Store, right? My point is showing a prompt to go to GitHub for users who downloaded from the App Store is confusing and backwards.

[Bundle](https://developer.apple.com/documentation/foundation/bundle/1407276-appstorereceipturl) has property `appStoreReceiptURL`

> For an application purchased from the App Store, use this application bundle property to locate the receipt. This property makes no guarantee about whether there is a file at the URL—only that if a receipt is present, that is its location.

Reply:
updated https://github.com/Sequel-Ace/Sequel-Ace/tree/github-release-checker a bit

Reply:
That looks super cool, @jamesstout! One thing - could we make it switch on allowing/not allowing pre-release depending on if the app bundle id has the beta component? Basically, if you open beta you'd get beta update notices (and I guess release notices too, which is fine), but if you're on the latest release you'd only get release notices? That branch looks awesome though and seems like it would be very helpful to GitHub-based users

Reply:
In addition to your switch, I'd like to do a few things:

- progress indicator if they clicked Download (with ability to cancel)
- I added [Alamofire](https://github.com/Alamofire/Alamofire), which is a huge library, just for two network calls, I'd like to get rid of it, unless anyone thinks it's worth keeping? (it's a 5MB lib, just for one arch)

Reply:
See: #879